compiler/rustc_middle/src/ty/context.rs

```diff
@@ -89,6 +89,12 @@ pub trait OnDiskCache<'tcx>: rustc_data_structures::sync::Sync {
 #[derive(TyEncodable, TyDecodable, HashStable)]
 pub struct DelaySpanBugEmitted(());
 
+impl DelaySpanBugEmitted {
+    pub fn into_error_reported(self) -> ErrorReported {
+        ErrorReported::new_i_know_what_im_doing()
```

Comment:
This doesn't necessarily seem correct, with the current name of `ErrorReported`.
Though I think what needs to happen is `ErrorReported` should probably be renamed to e.g. `ErrorGuaranteed` or `CompilationWillFail` (though that feels too long).

Either way, `DelaySpanBugEmitted` should contain `ErrorReported` to avoid this call to the unchecked constructor.